bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,106 @@
+import abc
+import re
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.results.Result import Result
+from coalib.misc.Shell import run_shell_command
+from collections import namedtuple
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'commit_info',
+               'message')
+@generate_ordering('commit_info',
+                   'contents',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin, commit_info):
+        Result.__init__(self, origin, 'HEAD commit information')
+        self.commit_info = commit_info
+
+
+class VCSCommitBear(GlobalBear):
+    __metaclass__ = abc.ABCMeta
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    @abc.abstractmethod
+    def get_head_commit(self):
+        """
+        Return the commit message from the head commit
+        """
+
+    def analyze_git_commit(self, head_commit):
+        commit_info = namedtuple('Commit_Info',
+                                 'raw_commit_message commit_sha commit_type '
+                                 'modified_files added_files deleted_files')
+
+        raw_commit_message = head_commit
+
+        commit_sha = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+
+        commit_type = []
+
+        head_commit = head_commit.strip('\n')
+
+        ciskip_regex = re.compile(r'\[ci skip\]|\[skip ci\]')
+        match = re.search(ciskip_regex, head_commit)
+        if match:
```

Comment:
This can be simplified by just using one-line `re.match('(\[ci skip\]|\[skip ci\])', head_commit)`

Reply:
re.match is used only to match the full-string, that's why re.search instead

Reply:
yes, you can still shrink it into one line `if re.search('(\[ci skip\]|\[skip ci\])', head_commit):`